### DISCUSSION TITLE
Strange issue with Select broken position after v1.1.1 update

### DISCUSSION THREAD

**Discussion by datvm** [material-components/material-web#5350] [Q&A] [answered]
So after updating to v1.1.1, somehow 2nd `md-select` on this page is broken https://blazorwebmaterial.lukevo.com/select :

- Click the first select on the left hand side, it opens correctly.
- Click the second select on the right hand side, the menu opens at this location instead. This issue doesn't happen until you open the left hand side select.

![image](https://github.com/material-components/material-web/assets/6388546/cad1300f-bb24-425b-afb2-b963e8c153d5)

Strangely enough, I cannot reproduce it at all on Lit Playground even when copying the full HTML code over so I am not sure what the actual problem is and the above page is the only place I encounter this issue. Also it only happens to Chrome and Chromium Edge. On Firefox it works correctly. Any direction on how to pinpoint the issue?

The HTML snippet rendered is [Lit Playground (not reproducible there)](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQgXCJAbWF0ZXJpYWwvd2ViL2FsbFwiIn0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxoZWFkPlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL3NpbXBsZS1ncmVldGluZy5qc1wiPjwvc2NyaXB0PlxuICBcbiAgPGxpbmsgaHJlZj1cImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vYm9vdHN0cmFwQDUuMy4xL2Rpc3QvY3NzL2Jvb3RzdHJhcC5taW4uY3NzXCIgcmVsPVwic3R5bGVzaGVldFwiIGludGVncml0eT1cInNoYTM4NC00YncrL2FlcFAvWUM5NGhFcFZOVmdpWmRnSUM1K1ZLTkJRTkdDSGVLUlFOK1B0bW9IREVYdXBwdm5ESnpRSXU5XCIgY3Jvc3NvcmlnaW49XCJhbm9ueW1vdXNcIj5cbiAgPGxpbmsgcmVsPVwic3R5bGVzaGVldFwiIGhyZWY9XCJodHRwczovL2ZvbnRzLmdvb2dsZWFwaXMuY29tL2ljb24_ZmFtaWx5PU1hdGVyaWFsK1N5bWJvbHMrT3V0bGluZWRcIj5cbjwvaGVhZD5cbjxib2R5PlxuPGRpdiBjbGFzcz1cInJvd1wiPlxuICA8ZGl2IGNsYXNzPVwiY29sXCI-XG4gICAgPCEtLSEtLT5cbiAgICA8IS0tIS0tPlxuICAgIDxtZC1vdXRsaW5lZC1zZWxlY3QgY2xhc3M9XCJ3LTEwMFwiIHZhbHVlPVwiMy4xNDE1OTI2NTM1ODk3OTNcIiByZXF1aXJlZD1cIlwiIGxhYmVsPVwiU2VsZWN0IGEgY29uc3RhbnRcIiBzdXBwb3J0aW5nLXRleHQ9XCJDdXN0b20gdmFsaWRpdHkgb25seSBhY2NlcHRzIHZhbHVlcyBzbWFsbGVyIHRoYW4gNFwiIGhhcy1sZWFkaW5nLWljb249XCJcIiBzZWxlY3RlZC1pbmRleD1cIjBcIiByb2xlPVwicHJlc2VudGF0aW9uXCIgX2JsXzk9XCJcIj5cbiAgICAgIDwhLS0hLS0-XG4gICAgICA8bWQtaWNvbiBzbG90PVwibGVhZGluZy1pY29uXCIgYXJpYS1oaWRkZW49XCJ0cnVlXCIgX2JsXzE1PVwiXCI-ZnVuY3Rpb248L21kLWljb24-XG4gICAgICA8IS0tIS0tPlxuICAgICAgPG1kLXNlbGVjdC1vcHRpb24gdmFsdWU9XCIzLjE0MTU5MjY1MzU4OTc5M1wiIHJvbGU9XCJwcmVzZW50YXRpb25cIiBtZC1tZW51LWl0ZW09XCJcIiBfYmxfMTY9XCJcIiB0YWJpbmRleD1cIjBcIiBhcmlhLXNlbGVjdGVkPVwidHJ1ZVwiIHNlbGVjdGVkPVwiXCIgY2xhc3M9XCJcIj5cbiAgICAgICAgPGRpdiBzbG90PVwiaGVhZGxpbmVcIj5QaTwvZGl2PlxuICAgICAgICA8IS0tIS0tPlxuICAgICAgICA8ZGl2IHNsb3Q9XCJzdGFydFwiPs-APC9kaXY-XG4gICAgICAgIDwhLS0hLS0-XG4gICAgICAgIDxkaXYgc2xvdD1cIm92ZXJsaW5lXCI-My4xNDE1OTI2NTM1ODk3OTM8L2Rpdj5cbiAgICAgICAgPCEtLSEtLT5cbiAgICAgICAgPGRpdiBzbG90PVwic3VwcG9ydGluZy10ZXh0XCI-VGhlIHJhdGlvIG9mIGEgY2lyY2xlJ3MgY2lyY3VtZmVyZW5jZSB0byBpdHMgZGlhbWV0ZXI8L2Rpdj5cbiAgICAgICAgPCEtLSEtLT5cbiAgICAgICAgPCEtLSEtLT5cbiAgICAgICAgPCEtLSEtLT5cbiAgICAgICAgPG1kLWljb24tYnV0dG9uIHNsb3Q9XCJlbmRcIiBocmVmPVwiaHR0cHM6Ly9lbi53aWtpcGVkaWEub3JnL3dpa2kvUGlcIiB0YXJnZXQ9XCJfYmxhbmtcIiByb2xlPVwicHJlc2VudGF0aW9uXCIgX2JsXzQxPVwiXCI-XG4gICAgICAgICAgPCEtLSEtLT5cbiAgICAgICAgICA8bWQtaWNvbiBhcmlhLWhpZGRlbj1cInRydWVcIiBfYmxfNDQ9XCJcIj5vcGVuX2luX25ldzwvbWQtaWNvbj5cbiAgICAgICAgPC9tZC1pY29uLWJ1dHRvbj5cbiAgICAgIDwvbWQtc2VsZWN0LW9wdGlvbj5cbiAgICAgIDwhLS0hLS0-XG4gICAgICA8bWQtc2VsZWN0LW9wdGlvbiB2YWx1ZT1cIjIuNzE4MjgxODI4NDU5MDQ1XCIgcm9sZT1cInByZXNlbnRhdGlvblwiIG1kLW1lbnUtaXRlbT1cIlwiIF9ibF8xNz1cIlwiIHRhYmluZGV4PVwiLTFcIiBjbGFzcz1cIlwiPlxuICAgICAgICA8ZGl2IHNsb3Q9XCJoZWFkbGluZVwiPkV1bGVyJ3MgbnVtYmVyPC9kaXY-XG4gICAgICAgIDwhLS0hLS0-XG4gICAgICAgIDxkaXYgc2xvdD1cInN0YXJ0XCI-ZTwvZGl2PlxuICAgICAgICA8IS0tIS0tPlxuICAgICAgICA8ZGl2IHNsb3Q9XCJvdmVybGluZVwiPjIuNzE4MjgxODI4NDU5MDQ1PC9kaXY-XG4gICAgICAgIDwhLS0hLS0-XG4gICAgICAgIDxkaXYgc2xvdD1cInN1cHBvcnRpbmctdGV4dFwiPlRoZSBiYXNlIG9mIG5hdHVyYWwgbG9nYXJpdGhtczwvZGl2PlxuICAgICAgICA8IS0tIS0tPlxuICAgICAgICA8IS0tIS0tPlxuICAgICAgICA8IS0tIS0tPlxuICAgICAgICA8bWQtaWNvbi1idXR0b24gc2xvdD1cImVuZFwiIGhyZWY9XCJodHRwczovL2VuLndpa2lwZWRpYS5vcmcvd2lraS9FXyhtYXRoZW1hdGljYWxfY29uc3RhbnQpXCIgdGFyZ2V0PVwiX2JsYW5rXCIgcm9sZT1cInByZXNlbnRhdGlvblwiIF9ibF80Mj1cIlwiPlxuICAgICAgICAgIDwhLS0hLS0-XG4gICAgICAgICAgPG1kLWljb24gYXJpYS1oaWRkZW49XCJ0cnVlXCIgX2JsXzQ1PVwiXCI-b3Blbl9pbl9uZXc8L21kLWljb24-XG4gICAgICAgIDwvbWQtaWNvbi1idXR0b24-XG4gICAgICA8L21kLXNlbGVjdC1vcHRpb24-XG4gICAgICA8IS0tIS0tPlxuICAgICAgPG1kLXNlbGVjdC1vcHRpb24gdmFsdWU9XCI2LjI4MzE4NTMwNzE3OTU4NlwiIHJvbGU9XCJwcmVzZW50YXRpb25cIiBtZC1tZW51LWl0ZW09XCJcIiBfYmxfMTg9XCJcIiB0YWJpbmRleD1cIi0xXCIgY2xhc3M9XCJcIj5cbiAgICAgICAgPGRpdiBzbG90PVwiaGVhZGxpbmVcIj5UYXU8L2Rpdj5cbiAgICAgICAgPCEtLSEtLT5cbiAgICAgICAgPGRpdiBzbG90PVwic3RhcnRcIj7PhDwvZGl2PlxuICAgICAgICA8IS0tIS0tPlxuICAgICAgICA8ZGl2IHNsb3Q9XCJvdmVybGluZVwiPjYuMjgzMTg1MzA3MTc5NTg2PC9kaXY-XG4gICAgICAgIDwhLS0hLS0-XG4gICAgICAgIDxkaXYgc2xvdD1cInN1cHBvcnRpbmctdGV4dFwiPlJhdGlvIG9mIGEgY2lyY2xlJ3MgY2lyY3VtZmVyZW5jZSB0byBpdHMgcmFkaXVzPC9kaXY-XG4gICAgICAgIDwhLS0hLS0-XG4gICAgICAgIDwhLS0hLS0-XG4gICAgICAgIDwhLS0hLS0-XG4gICAgICAgIDxtZC1pY29uLWJ1dHRvbiBzbG90PVwiZW5kXCIgaHJlZj1cImh0dHBzOi8vZW4ud2lraXBlZGlhLm9yZy93aWtpL1R1cm5fKGFuZ2xlKSNUYXVfcHJvcG9zYWxzXCIgdGFyZ2V0PVwiX2JsYW5rXCIgcm9sZT1cInByZXNlbnRhdGlvblwiIF9ibF80Mz1cIlwiPlxuICAgICAgICAgIDwhLS0hLS0-XG4gICAgICAgICAgPG1kLWljb24gYXJpYS1oaWRkZW49XCJ0cnVlXCIgX2JsXzQ2PVwiXCI-b3Blbl9pbl9uZXc8L21kLWljb24-XG4gICAgICAgIDwvbWQtaWNvbi1idXR0b24-XG4gICAgICA8L21kLXNlbGVjdC1vcHRpb24-XG4gICAgPC9tZC1vdXRsaW5lZC1zZWxlY3Q-XG4gICAgPCEtLSEtLT5cbiAgICA8cD5TZWxlY3RlZCB2YWx1ZTogMy4xNDE1OTI2NTM1ODk3OTMsIHNlbGVjdGVkIGluZGV4OiAwPC9wPlxuICA8L2Rpdj5cbiAgPCEtLSEtLT5cbiAgPGRpdiBjbGFzcz1cImNvbFwiPlxuICAgIDwhLS0hLS0-XG4gICAgPCEtLSEtLT5cbiAgICA8bWQtZmlsbGVkLXNlbGVjdCBjbGFzcz1cInctMTAwXCIgdmFsdWU9XCJBcHBsZVwiIHJlcXVpcmVkPVwiXCIgbGFiZWw9XCJTZWxlY3QgYSBmcnVpdFwiIHN1cHBvcnRpbmctdGV4dD1cIk1heSBiZSBlbXB0eVwiIGhhcy1sZWFkaW5nLWljb249XCJcIiBzZWxlY3RlZC1pbmRleD1cIjBcIiByb2xlPVwicHJlc2VudGF0aW9uXCIgX2JsXzEwPVwiXCI-XG4gICAgICA8IS0tIS0tPlxuICAgICAgPG1kLWljb24gc2xvdD1cImxlYWRpbmctaWNvblwiIGFyaWEtaGlkZGVuPVwidHJ1ZVwiIF9ibF8xOT1cIlwiPm51dHJpdGlvbjwvbWQtaWNvbj5cbiAgICAgIDwhLS0hLS0-XG4gICAgICA8bWQtc2VsZWN0LW9wdGlvbiB2YWx1ZT1cIlwiIHJvbGU9XCJwcmVzZW50YXRpb25cIiBtZC1tZW51LWl0ZW09XCJcIiBfYmxfMjA9XCJcIiB0YWJpbmRleD1cIi0xXCIgY2xhc3M9XCJtZC1tZW51LWhpZGRlblwiPk5vIHRoYW5rczwvbWQtc2VsZWN0LW9wdGlvbj5cbiAgICAgIDwhLS0hLS0-XG4gICAgICA8IS0tIS0tPlxuICAgICAgPG1kLWRpdmlkZXIgX2JsXzIxPVwiXCI-PC9tZC1kaXZpZGVyPlxuICAgICAgPCEtLSEtLT5cbiAgICAgIDxtZC1zZWxlY3Qtb3B0aW9uIHNlbGVjdGVkPVwiXCIgdmFsdWU9XCJBcHBsZVwiIHJvbGU9XCJwcmVzZW50YXRpb25cIiBtZC1tZW51LWl0ZW09XCJcIiBfYmxfMjI9XCJcIiBhcmlhLXNlbGVjdGVkPVwidHJ1ZVwiIHRhYmluZGV4PVwiLTFcIiBjbGFzcz1cIlwiPlxuICAgICAgICA8ZGl2IHNsb3Q9XCJoZWFkbGluZVwiPkFwcGxlPC9kaXY-XG4gICAgICA8L21kLXNlbGVjdC1vcHRpb24-XG4gICAgICA8IS0tIS0tPlxuICAgICAgPG1kLXNlbGVjdC1vcHRpb24gdmFsdWU9XCJCYW5hbmFcIiByb2xlPVwicHJlc2VudGF0aW9uXCIgbWQtbWVudS1pdGVtPVwiXCIgX2JsXzIzPVwiXCIgdGFiaW5kZXg9XCItMVwiIGNsYXNzPVwiXCI-XG4gICAgICAgIDxkaXYgc2xvdD1cImhlYWRsaW5lXCI-QmFuYW5hPC9kaXY-XG4gICAgICA8L21kLXNlbGVjdC1vcHRpb24-XG4gICAgICA8IS0tIS0tPlxuICAgICAgPG1kLXNlbGVjdC1vcHRpb24gdmFsdWU9XCJDdWN1bWJlclwiIHJvbGU9XCJwcmVzZW50YXRpb25cIiBtZC1tZW51LWl0ZW09XCJcIiBfYmxfMjQ9XCJcIiB0YWJpbmRleD1cIi0xXCIgY2xhc3M9XCJcIj5cbiAgICAgICAgPGRpdiBzbG90PVwiaGVhZGxpbmVcIj5DdWN1bWJlcjwvZGl2PlxuICAgICAgPC9tZC1zZWxlY3Qtb3B0aW9uPlxuICAgIDwvbWQtZmlsbGVkLXNlbGVjdD5cbiAgICA8IS0tIS0tPlxuICAgIDxwPlNlbGVjdGVkIFN0cmluZyB2YWx1ZTogQXBwbGUsIHNlbGVjdGVkIGluZGV4OiAwPC9wPlxuICAgIDwhLS0hLS0-XG4gICAgPHA-XG4gICAgICA8IS0tIS0tPlxuICAgICAgPGNvZGU-TWRTZWxlY3RPcHRpb248L2NvZGU-IHNlbGVjdGVkOiBUcnVlLCBGYWxzZSwgRmFsc2VcbiAgICA8L3A-XG4gIDwvZGl2PlxuPC9kaXY-XG4gIFxuPC9ib2R5PlxuIn0seyJuYW1lIjoicGFja2FnZS5qc29uIiwiY29udGVudCI6IntcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwibGl0XCI6IFwiXjMuMC4wXCIsXG4gICAgXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjogXCJeMi4wLjBcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjQuMC4wXCIsXG4gICAgXCJsaXQtaHRtbFwiOiBcIl4zLjAuMFwiXG4gIH1cbn0iLCJoaWRkZW4iOnRydWV9XQ):

```html
<div class="row">
  <div class="col">
    <!--!-->
    <!--!-->
    <md-outlined-select class="w-100" value="3.141592653589793" required="" label="Select a constant" supporting-text="Custom validity only accepts values smaller than 4" has-leading-icon="" selected-index="0" role="presentation" _bl_9="">
      <!--!-->
      <md-icon slot="leading-icon" aria-hidden="true" _bl_15="">function</md-icon>
      <!--!-->
      <md-select-option value="3.141592653589793" role="presentation" md-menu-item="" _bl_16="" tabindex="0" aria-selected="true" selected="" class="">
        <div slot="headline">Pi</div>
        <!--!-->
        <div slot="start">π</div>
        <!--!-->
        <div slot="overline">3.141592653589793</div>
        <!--!-->
        <div slot="supporting-text">The ratio of a circle's circumference to its diameter</div>
        <!--!-->
        <!--!-->
        <!--!-->
        <md-icon-button slot="end" href="https://en.wikipedia.org/wiki/Pi" target="_blank" role="presentation" _bl_41="">
          <!--!-->
          <md-icon aria-hidden="true" _bl_44="">open_in_new</md-icon>
        </md-icon-button>
      </md-select-option>
      <!--!-->
      <md-select-option value="2.718281828459045" role="presentation" md-menu-item="" _bl_17="" tabindex="-1" class="">
        <div slot="headline">Euler's number</div>
        <!--!-->
        <div slot="start">e</div>
        <!--!-->
        <div slot="overline">2.718281828459045</div>
        <!--!-->
        <div slot="supporting-text">The base of natural logarithms</div>
        <!--!-->
        <!--!-->
        <!--!-->
        <md-icon-button slot="end" href="https://en.wikipedia.org/wiki/E_(mathematical_constant)" target="_blank" role="presentation" _bl_42="">
          <!--!-->
          <md-icon aria-hidden="true" _bl_45="">open_in_new</md-icon>
        </md-icon-button>
      </md-select-option>
      <!--!-->
      <md-select-option value="6.283185307179586" role="presentation" md-menu-item="" _bl_18="" tabindex="-1" class="">
        <div slot="headline">Tau</div>
        <!--!-->
        <div slot="start">τ</div>
        <!--!-->
        <div slot="overline">6.283185307179586</div>
        <!--!-->
        <div slot="supporting-text">Ratio of a circle's circumference to its radius</div>
        <!--!-->
        <!--!-->
        <!--!-->
        <md-icon-button slot="end" href="https://en.wikipedia.org/wiki/Turn_(angle)#Tau_proposals" target="_blank" role="presentation" _bl_43="">
          <!--!-->
          <md-icon aria-hidden="true" _bl_46="">open_in_new</md-icon>
        </md-icon-button>
      </md-select-option>
    </md-outlined-select>
    <!--!-->
    <p>Selected value: 3.141592653589793, selected index: 0</p>
  </div>
  <!--!-->
  <div class="col">
    <!--!-->
    <!--!-->
    <md-filled-select class="w-100" value="Apple" required="" label="Select a fruit" supporting-text="May be empty" has-leading-icon="" selected-index="0" role="presentation" _bl_10="">
      <!--!-->
      <md-icon slot="leading-icon" aria-hidden="true" _bl_19="">nutrition</md-icon>
      <!--!-->
      <md-select-option value="" role="presentation" md-menu-item="" _bl_20="" tabindex="-1" class="md-menu-hidden">No thanks</md-select-option>
      <!--!-->
      <!--!-->
      <md-divider _bl_21=""></md-divider>
      <!--!-->
      <md-select-option selected="" value="Apple" role="presentation" md-menu-item="" _bl_22="" aria-selected="true" tabindex="-1" class="">
        <div slot="headline">Apple</div>
      </md-select-option>
      <!--!-->
      <md-select-option value="Banana" role="presentation" md-menu-item="" _bl_23="" tabindex="-1" class="">
        <div slot="headline">Banana</div>
      </md-select-option>
      <!--!-->
      <md-select-option value="Cucumber" role="presentation" md-menu-item="" _bl_24="" tabindex="-1" class="">
        <div slot="headline">Cucumber</div>
      </md-select-option>
    </md-filled-select>
    <!--!-->
    <p>Selected String value: Apple, selected index: 0</p>
    <!--!-->
    <p>
      <!--!-->
      <code>MdSelectOption</code> selected: True, False, False
    </p>
  </div>
</div>
```

**Comment by asyncliz** [comment]
Hm I'm not able to reproduce on Chrome v120 on Mac OS when I go to your site. What browser and OS are you seeing the issues on?

**Reply by datvm** [reply]
I tested it on Windows 11. Strange, I will test it on my Mac later and see what happens. Probably only happens on Windows then.

**Reply by asyncliz** [reply]
I see it on Windows 11 too. @e111077 might have some inset when they get back

**Reply by datvm** [reply]
Thanks, will wait. I tested on the Mac and it doesn't happen on Mac with Chromium Edge as well. Strange indeed.

**Comment by e111077** [comment]
when we are calculating the size of the menu, we invisibly render it on the page. In popover mode, since it's top-layer, it renders at the end of the page. It seems that this is causing your page to relayout for a frame in which the second select is rendered below the first one (compact view mode). Then it calculates where it's supposed to render from the wrong spot. I think the answer here is to set `position: absolute; inset-block-start:0;` on the menu in this step to make sure that it tries its best to not resize the page when calculating the natural height of the menu

**Reply by e111077** [reply]
also is there something going on with that page's compact / wide layout? it seems to get stuck in one or the other

**Reply by datvm** [reply]
Hi, sorry about the late response, late and long holiday for me 😅. It's strange indeed. I didn't notice the compact/wide layout affect the column layout as well. I am not really sure what is happening there...

**Reply by datvm** [reply, marked as answer]
@e111077 Not sure what you fixed but after updating to v1.2.0, the issue is not there anymore, both issues: https://blazorwebmaterial.lukevo.com/select

**Comment by shadow-identity** [comment]
I have a similar issue with select menu positioning on Firefox 115 (ESR) and 121 (current), and what is especially weird, is that in vite dev server there is no issues, but production build has it. The same code, the same browser, but different way of building.

I found that positioning is broken if: 
- absolutely unrelated `md-menu` element has `positioning=popover`
- `menu-positioning="fixed"` is used on the `<select>`
- production build
- Firefox < 122 beta

I was able to fix it by removing `positioning=popover` from the top menu T_T and removing `menu-positioning="fixed"` from the select.

I think it worth it to report a bug to investigate the inconsistency.

UPD: with removed `menu-positioning="fixed"` from the `<select>` the production build was working on a local machine but not on the deployment server. To make it finally work I was needed to add explicit `menu-positioning="absolute"` 🤪
